handlers/tokeninfo/jwt/tokeninfo.go

```diff
@@ -120,23 +128,35 @@ func newTokenInfo(t *jwt.Token, timeBase time.Time) (*TokenInfo, error) {
 	}, nil
 }
 
-func claimAsStrings(t *jwt.Token, claim string) ([]string, bool) {
+func NewTokenInfo(t *jwt.Token, timeBase time.Time) (*TokenInfo, error) {
+	issuer, ok := ClaimAsString(t, JwtClaimIssuer)
+	if ok {
+		processor, found := options.AppSettings.JwtProcessors[issuer]
+		if found {
+			log.Printf("token matched processor for issuer %s", issuer)
+			return processor.(JwtProcessor).Process(t, timeBase)
+		}
+	}
+	return defaultNewTokenInfo(t, timeBase)
+}
+
+func ClaimAsStrings(t *jwt.Token, claim string) ([]string, bool) {
 	if c, ok := getClaim(t, claim); ok {
 		value, ok := c.([]interface{})
 		if !ok {
 			log.Printf("Invalid string array value for claim %q = %v", claim, c)
 			return nil, false
 		}
-		strings := make([]string, len(value))
+		result:= make([]string, len(value))
 		for i, scope := range value {
-			strings[i] = scope.(string)
+			result[i] = scope.(string)
 		}
-		return strings, true
+		return result, true
 	}
 	return nil, false
 }
 
-func claimAsString(t *jwt.Token, claim string) (string, bool) {
+func ClaimAsString(t *jwt.Token, claim string) (string, bool) {
```

Comment:
@vroldanbet Yeah I would never duplicate the code, but if it's a utility function then it could be in a dedicated sub package. But where it is now makes sense given the context :)